src/ng/directive/ngRef.js

```diff
@@ -0,0 +1,220 @@
+'use strict';
+
+/**
+ * @ngdoc directive
+ * @name ngRef
+ * @restrict A
+ *
+ * @description
+ * The `ngRef` attribute tells AngularJS to assign the controller of a component (or an element-directive)
+ * to the given property in the current scope. If no controller exists, the jqlite-wrapped DOM
+ * element will be added to the scope.
+ *
+ * If the element with ngRef is destroyed `null` is assigned to the property.
+ *
+ *
+ * @element ANY
+ * @param {string} ngRef property name - A valid AngularJS expression identifier to which the
+ *                       controller or jqlite-wrapped DOM element will be bound.
+ * @param {string=} ngRefElement void - If specified, ngRef will always bind the jqlite-wrapped
+ *                               DOM-element even if a controller is available.
```

Comment:
What does the `void -` bit mean in the description here?

Reply:
It is supposed to signify that the attribute takes no value. :)